src/Compilers/CSharp/Test/Symbol/Symbols/ConversionTests.cs

```diff
@@ -425,6 +426,112 @@ static void Main()
             Assert.True(conversion.IsNumeric);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/36377")]
+        public void GetSymbolInfo_ExplicitCastOnMethodGroup()
+        {
+            var src = """
+public sealed class C
+{
+    public static void M()
+    {
+        C x = (C)C.Test;
+    }
+
+    public static int Test() => 1;
+
+    public static explicit operator C(System.Func<int> intDelegate)
```

Comment:
>explicit [](http://example.com/codeflow?start=18&length=8)

Consider testing with implicit operator, including a scenario when the cast operator, `(C)`,  is not present in code.

Reply:
Done. There are some existing unexpected behaviors there. Filing issues for those